src/samples/audio/detect_test.c

```diff
@@ -72,6 +73,9 @@ struct comp_data {
 	struct kpb_event_data event_data;
 	struct kpb_client client_data;
 
+	int16_t *input;
+	size_t input_size;
+
```

Comment:
Traditionally this is called `local_buffer` and it is of the type `struct comp_buffer`? Could be this case for you too?

Check `struct host_data` in `src/audio/host.c`

Reply:
comp_buffer doesn't have the support to convert to a linear buffer, so it's not exactly adequate. (the support I wanted to add at some point was scrapped)

Reply:
I think this one can be adapter https://github.com/thesofproject/sof/blob/master/src/audio/codec_adapter/codec_adapter.c#L297